src/adams_methods.jl

```diff
@@ -0,0 +1,655 @@
+                                            -1//12    8//12    5//12     0
+                                            1//24     -5//24   19//24    9//24]
+
+
+
+###############################################################################
+# Gamma star function from Hairer et al pg 359, table 1.2
+# used in adaptive step selection (See Harier pg 423)
+###############################################################################
+"compute γ*(j) for ode113"
+function γ_star(j)
+    if j == 0
+        return 1
+    elseif j > 0
+
+    return Polynomials.polyval((1 / factorial(j))*
+            Polynomials.polyint(Polynomials.poly(diagm(collect(1-(j-1):1)))),1)
+    else
+        error("order must be positive integer")
+    end
+end
+
+################################################################################
+# Helper functions for calculating coefficients from recurence relationships
+################################################################################
+#Based on Harier Lemma 5.1 on page 399
+#Calculates c_j,q and g_n,j for n and j =1:steporder
+"compute g_{n,j} coefficients for ode113"
+function g_coefs!(g,c,h,t,t_np1,n,steporder)
+    for j = 1:steporder
+      for q = 1:(steporder)-(j-1)
+        if j ==1
+          c[j,q]=1/q
+        elseif j==2
+          c[j,q]=1/q/(q+1)
+        else
+          c[j,q] = c[j-1,q] - c[j-1,q+1]*h/(t_np1-t[n-(j-1)+1])
+        end
+      end
+      g[j]=c[j,1]
+    end
+end
+
+#Based on Harier equation 5.9. Calculates b, ϕ, ϕstar for
+#n and j  in range_of_indices
+"compute ϕ_{n,j} and ϕ*_{n,j} coefficients for ode113"
+function ϕ_and_ϕstar_coefs!(b, ϕ, ϕstar,dy_n,t,t_np1,n,range_of_indices)
+    for j in range_of_indices #0:k-1
+      if j== 1#0
+        b[j] = 1
+        ϕ[2,j] = dy_n
+        ϕstar[2,j] = dy_n
+      else j>1#0
+        #Note: since n>=j, when this runs, n>=2 (so n-1>=1) and
+        b[j] = b[j-1]*(t_np1-t[n-(j-1)+1])/(t[n]-t[n-(j-1)])
+        ϕ[2,j] =  ϕ[2,j-1]- ϕstar[1,j-1]
+        ϕstar[2,j] = b[j]*ϕ[2,j]
+      end
+    end
+end
+
+
+#Based on Harier equation 5.9. Just calculates the ϕ coefficients
+"compute ϕ_{n+1,j} coefficients for ode113"
+function ϕ_coefs!(ϕ, ϕstar,dy_n,range_of_indices)
+    for j  in range_of_indices
+      if j== 1#0
+        ϕ[3,j] = dy_n
+      else j>1#0
+        #Note: since n>=j, when this runs, n>=2 (so n-1>=1) and
+        ϕ[3,j] =  ϕ[3,j-1]- ϕstar[2,j-1]
+      end
+    end
+end
+
+################################################################################
+# Helper functions initializing/resizing arrays
+################################################################################
+"used to initialize working arrays used in ode1113"
+function intialize_coef_arrays(dim,T)
+    c = zeros(dim,dim)
+    g = zeros(dim)
+    b = zeros(dim)
+    @compat ϕstar = Array{T,2}(2,dim)
+    @compat ϕ = Array{T,2}(3,dim)
+    return c,g,b,ϕstar, ϕ
+end
+
+
+function make_consistent_types(fn, y0, tspan)
```

Comment:
It would be good to have this refactored with the existing one: just make the one taking a `btab` too call this one.  ((Almost) never copy-paste code).  Also move to ODE.jl-file.